mixer/adapter/svcctrl/reportbuilder_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package svcctrl
+
+import (
+	"testing"
+	"time"
+
+	sc "google.golang.org/api/servicecontrol/v1"
+
+	"istio.io/istio/mixer/adapter/svcctrl/template/svcctrlreport"
+)
+
+func getTestReportBuilder() *reportBuilder {
+	requestTime, _ := time.Parse(time.RFC3339Nano, "2017-10-21T17:09:05.000Z")
+	responseTime, _ := time.Parse(time.RFC3339Nano, "2017-10-21T17:09:05.100Z")
+
+	return &reportBuilder{
+		supportedMetrics: []metricDef{
+			{
+				name:           "test_metric",
+				valueGenerator: generateRequestCount,
+				labels: []string{
+					"/consumer_id",
+					"/credential_id",
+					"/error_type",
+					"/protocol",
+					"/response_code",
+					"/response_code_class",
+					"/status_code",
+				},
+			},
+		},
+		instance: &svcctrlreport.Instance{
+			ApiVersion:      "v1.0",
+			ApiOperation:    "POST /echo",
+			ApiProtocol:     "REST",
+			ApiService:      "echo.test.com",
+			ApiKey:          "test_key",
+			RequestTime:     requestTime,
+			RequestMethod:   "POST",
+			RequestPath:     "echo.test.com/echo",
+			RequestBytes:    10,
+			ResponseTime:    responseTime,
+			ResponseCode:    200,
+			ResponseBytes:   1024,
+			ResponseLatency: 1 * time.Microsecond,
+		},
+		resolver: &mockConsumerProjectIDResolver{
+			consumerProjectID: "test_consumer_project",
+		},
+		labelMap: make(map[string]string),
+	}
+}
+
+func TestBuildLogEntry(t *testing.T) {
+	rb := getTestReportBuilder()
+	op := &sc.Operation{}
+	rb.addLogEntry(op)
+
+	if len(op.LogEntries) != 1 {
+		t.Errorf(`len(op.LogEntries) != 1`)
+	}
+
+	expected :=
+		`{
+			"api_name":"echo.test.com",
+			"api_method":"POST /echo",
```

Comment:
change all api_method to api_operation in Istio context. Only use api_method when sending ReportRequest.

Let's use  "echo" as an example, instead of "POST / echo".

Reply:
done